crates/compiler/can/src/def.rs

```diff
@@ -2880,6 +2941,188 @@ fn to_pending_value_def<'a>(
+                None => {
+                    exposed_symbols = Vec::new();
+
+                    if is_automatically_imported {
+                        env.problems
+                            .push(Problem::ExplicitBuiltinImport(module_id, region));
+                    }
+                }
+                Some(exposed_kw) => {
+                    let exposed_ids = env
+                        .dep_idents
+                        .get(&module_id)
+                        .expect("Module id should have been added in load");
+
+                    exposed_symbols = Vec::with_capacity(exposed_kw.item.len());
+
+                    for loc_name in exposed_kw.item.items {
+                        let exposed_name = loc_name.value.item();
+                        let name = exposed_name.as_str();
+                        let ident = Ident::from(name);
+
+                        match exposed_ids.get_id(name) {
+                            Some(ident_id) => {
+                                let symbol = Symbol::new(module_id, ident_id);
+                                exposed_symbols.push((symbol, loc_name.region));
+
+                                match scope.import_symbol(ident, symbol, loc_name.region) {
+                                    Ok(()) => {}
+                                    Err((_shadowed_symbol, existing_symbol_region)) => {
+                                        if is_automatically_imported
+                                            && Symbol::builtin_types_in_scope(module_id)
+                                                .iter()
+                                                .any(|(_, (s, _))| *s == symbol)
+                                        {
+                                            env.problem(Problem::ExplicitBuiltinTypeImport(
+                                                symbol,
+                                                loc_name.region,
+                                            ));
+                                        } else {
+                                            env.problem(Problem::ImportShadowsSymbol {
+                                                region: loc_name.region,
+                                                new_symbol: symbol,
+                                                existing_symbol_region,
+                                            })
+                                        }
+                                    }
+                                }
+                            }
+                            None => {
+                                let exposed_values = exposed_ids
+                                    .ident_strs()
+                                    .filter(|(_, ident)| {
+                                        ident.starts_with(|c: char| c.is_lowercase())
+                                    })
+                                    .map(|(_, ident)| Lowercase::from(ident))
+                                    .collect();
```

Comment:
This appears in a few other places as well:

https://github.com/roc-lang/roc/blob/5112e064e571275b4a53a6a45e4a8c70a2f959b9/crates/compiler/can/src/env.rs#L174-L178

I wonder if it should be abstracted behind a method on exposed_ids.

Reply:
Addressed [here](https://github.com/roc-lang/roc/pull/6658/commits/5ca1494df12280f8f6d4ab410d460a3d851b1016)